src/util/generateConnection.ts

```diff
@@ -0,0 +1,67 @@
+import * as fs from './fs';
+import { Key, Logger, Mapping } from './types';
+
+const CONNECTIONS_FILE_NAME = 'connections.sync-db.json';
+const KEYS = ['DB_HOST', 'DB_PASSWORD', 'DB_NAME', 'DB_USERNAME', 'DB_PORT', 'DB_CLIENT', 'DB_ENCRYPTION'];
+const mapping: Mapping<string> = {
+  USERNAME: 'name',
+  NAME: 'database'
+};
+
+/**
+ * Reads values from ENV from provide keys.
+ *
+ * @param {string[]} keys
+ * @returns {string[] | Key}
+ */
+function extractFromEnv(keys: string[]): string[] | Key {
+  const errors: string[] = [];
+  const connection: Key = {};
+
+  keys.forEach((element: string) => {
+
+    if (element === 'DB_ENCRYPTION') {
+      connection['options'] = {
+        encrypt: (process.env[element] === 'true') || false
+      };
+    } else if (!process.env[element]) {
+      errors.push(element);
+    } else {
+      const [, b] = element.split('_');
+      const key = mapping[b] ? mapping[b].toLowerCase() : b.toLowerCase();
+      connection[key] = String(process.env[element]);
+    }
+
+  });
+
+  if (errors.length) {
+    throw new Error(JSON.stringify({ 'keys not found': errors }));
+  }
+
+  return connection;
+}
```

Comment:
This implementation doesn't look neat enough. Look into some of the existing code here to get some idea:
- https://github.com/leapfrogtechnology/sync-db/blob/master/src/config.ts
- https://github.com/leapfrogtechnology/sync-db/blob/master/src/constants.ts

Some of the things you can reuse already. Also, make this function `extractFromEnv` return `Promise<ConnectionConfig[]>` i.e same as what [resolveConnections](https://github.com/leapfrogtechnology/sync-db/blob/master/src/config.ts#L37) returns.

Also, you can move this function to `config.ts` and name it `resolveConnectionsFromEnv` - because it's similar.

Reply:
Make it straightforward like the example code here in the issue https://github.com/leapfrogtechnology/sync-db/issues/31.

You can do:
```ts
const connection = {
  client: process.env.DB_CLIENT,
  host: process.env.DB_HOST,
  ...
  options: {
    encrypt: ...
  }
};
```

Reply:
Instead of
~~validateConnections(process.env)~~, `validateConnections(conn_keys)`. And access the `process.env` inside the function.

And Loop is compulsory to validate the connections? Will this implementation be Okay?